src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea() : number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (!this.checkTriangle()) {
+      throw new Error('There are no correct triangle sizes');
+    }
+  }
+
+  checkTriangle(): boolean {
```

Comment:
```suggestion
  isValidTriangle(): boolean {
```

Reply:
It is a cool method and I like it. Let's make it private, cause you are using it only internally